wire-schema/src/test/java/com/squareup/wire/schema/SchemaProtoAdapterTest.kt

```diff
@@ -0,0 +1,313 @@
+/*
+ * Copyright (C) 2015 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.wire.schema
+
+import com.google.common.collect.ImmutableList
+import com.google.common.collect.ImmutableMap
+import com.squareup.wire.ProtoAdapter
+import java.io.EOFException
+import java.io.IOException
+import java.net.ProtocolException
+import okio.Buffer
+import okio.ByteString
+import okio.ByteString.Companion.decodeHex
+import org.junit.Test
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Assert.fail
+
+class SchemaProtoAdapterTest {
+    private val coffeeSchema = RepoBuilder()
+            .add("coffee.proto",
+                    """
+                    message CafeDrink {
+                      optional string customer_name = 1;
+                      repeated EspressoShot shots = 2;
+                      optional Foam foam = 3;
+                      optional int32 size_ounces = 14;
+                      optional Dairy dairy = 15;
+
+                      enum Foam {
+                        NOT_FOAMY_AND_QUITE_BORING = 1;
+                        ZOMG_SO_FOAMY = 3;
+                      }
+                    }
+
+                    message Dairy {
+                      optional int32 count = 2;
+                      optional string type = 1;
+                    }
+
+                    message EspressoShot {
+                      optional string bean_type = 1;
+                      optional double caffeine_level = 2;
+                    }
+                    """.trimMargin()
```

Comment:
For `trimMargin()` to work you have to prefix each string with `|`. As is it'll preserve all the whitespace preceding each line. You can switch to `trimIndent()`, or add `|` and stick to `trimMargin()` for consistency.

Reply:
i think i started to space out here, from all the mechanical conversion 😸